docs/pattern-features.md

```diff
@@ -227,6 +228,29 @@ pattern: $X = 1 + 2 + ...
 foo = 1 + 2 + 3 + 4
 ```
 
+### Deep Expression Operator
+
+You may want to match an expression that could be nested deep within another expression. An example of this is looking for a pattern anywhere within an if statement. To do this, use the deep expression operator: `<... [your_pattern] ...>`. This will match your pattern in the current expression context and recursively in any subexpressions.
+
+For example, this pattern:
+
+```yaml
+if <... $USER.is_admin() ...>:
```

Comment:
Can we change this to the full `pattern` syntax? E.g.

```yaml
pattern: |
    if <... $USER.is_admin() ...>:
        ...
```

I think that will provide users with more context on where they would use this.